tests/test_views.py

```diff
@@ -974,6 +979,56 @@ def _login(self):
         """
         assert self.client.login(username=self.user.username, password="QWERTY")
 
+    def _setup_course_catalog_client(self, client_mock):
+        """
+        Sets up the Course Catalog API client
+        """
+        client = client_mock.return_value
+        client.get_course_run.return_value = {
+            'level_type': 'Type 1',
+        }
+
+    def _setup_organizations_client(self, client_mock, data=DUMMY_PLACEHOLDER, side_effect=None):
+        """
+        Sets up the Organizations API client
+        """
+        if side_effect is None:
+            if data is DUMMY_PLACEHOLDER:
+                logo_mock = mock.MagicMock()
+                logo_mock.url = 'logo.png'
+                data = {
+                    'logo': logo_mock,
+                    'name': 'Organization',
+                }
+            client_mock.get_organization.return_value = data
+        else:
+            client_mock.get_organization.side_effect = side_effect
+
+    def _setup_course_api_client(self, client_mock, data=DUMMY_PLACEHOLDER, side_effect=None):
+        """
+        Sets up the Courses API client
+        """
+        client = client_mock.return_value
+        if side_effect is None:
+            if data is DUMMY_PLACEHOLDER:
+                data = self.dummy_demo_course_details_data
+            client.get_course_details.return_value = data
+        else:
+            client.get_course_details.side_effect = HttpClientError
+
+    def _setup_enrollment_client(self, client_mock, data=DUMMY_PLACEHOLDER, side_effect=None):
```

Comment:
@replaceafill Similar comment here: There's 10 calls to this method but only a couple pass custom values for `data` and/or `side_effect`. You can remove these calls and specify custom behavior inline; that will let you get rid of the `data` and `side_effect` parameters here.

Ultimately this will also let you get rid of `DUMMY_PLACEHOLDER` altogether.